### PR TITLE
ocaml-lsp update

### PR DESCRIPTION
**Pull request by symphorien**
###### Description of changes

tested as:
`nix build -f. ocaml-ng.ocamlPackages_4_{06,07,08,09,10,11,12,13}.ocaml-lsp --keep-going`

and on a real project with ocaml 4.08, 12, and 13

4.14 ppxlib does not build so untested.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/ocaml-packages.nix [hunk 1]
```diff
@@ -324,8 +324,14 @@ let
       inherit (pkgs) opam git mercurial coreutils gnutar bzip2;
     };
 
+    dune-rpc = callPackage ../development/ocaml-modules/dune-rpc { };
+
     dune-site = callPackage ../development/ocaml-modules/dune-site { };
 
+    fiber = callPackage ../development/ocaml-modules/fiber { };
+
+    xdg = callPackage ../development/ocaml-modules/xdg { };
```
Comment (superherointj) [inline pkgs/top-level/ocaml-packages.nix:333]:
```suggestion
    fiber = callPackage ../development/ocaml-modules/fiber { };

    xdg = callPackage ../development/ocaml-modules/xdg { };
```

`fiber` and `xdg` are not in alphabetical order.

File: pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix [hunk 2]
```diff
@@ -10,10 +10,14 @@
 }:
 
 let params =
-  if lib.versionAtLeast ocaml.version "4.13"
+  if lib.versionAtLeast ocaml.version "4.14"
   then {
-    version = "1.10.3";
-    sha256 = "sha256-o6wQc7Byi5T0vbARF3LAq69/9wMkOZRQ6rcVa/rBUfE=";
+    version = "1.11.1";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:15]:
```suggestion
    version = "1.11.1";
```
Current version is: `1.11.3 `

Comment (superherointj) [inline pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:15]:
```suggestion
    version = "1.11.3";
```
`1.11.3` is released already.

File: pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix [hunk 3]
```diff
@@ -10,10 +10,14 @@
...
+    sha256 = "sha256-jqiqGB6wXSXtWyqOAS/zZNPa0ezHthQdeLtYI0gc1qQ=";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:16]:
```suggestion
    sha256 = "sha256-KlMFh05O04I0Xil2B+nL2hUxZw0jaDMUnI23oUwGyhs=";
```

File: pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix [hunk 4]
```diff
@@ -10,10 +10,14 @@
...
+  } else if lib.versionAtLeast ocaml.version "4.13"
+  then {
+    version = "1.10.5";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:19]:
```suggestion
    version = "1.10.6";
```
1.10.6 is released

File: pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix [hunk 5]
```diff
@@ -10,10 +10,14 @@
...
+    sha256 = "sha256-TeJS6t1ruWhWPvWNatrnSUWI6T17XKiosHLYizBDDcw=";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:20]:
```suggestion
    sha256 = "sha256-K0OzWMIZjn5S1Oxu92TX0K2IuFfwn8twFQQoURv+YQw=";
```

File: pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix [hunk 6]
```diff
@@ -10,10 +10,14 @@
...
   } else if lib.versionAtLeast ocaml.version "4.12"
   then {
     version = "1.9.0";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:23]:
```suggestion
    version = "1.9.1";
```
https://github.com/ocaml/ocaml-lsp/tree/1.9.1

Fixes
* Disable functionality reliant on ocamlformat-rpc for now (https://github.com/ocaml/ocaml-lsp/pull/555)
* 4.13 compatiblity

Comment (superherointj) [inline pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:23]:
```suggestion
    version = "1.9.1";
```
https://github.com/ocaml/ocaml-lsp/tree/1.9.1

Fixes:
* Disable functionality reliant on ocamlformat-rpc for now
* 4.13 compatiblity

Comment (superherointj) [inline pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:23]:
```suggestion
    version = "1.9.1";
    sha256 = "sha256-zxzbJX1O8e0aBelRJksRx9x9MGa/RIERyxfGe+Vt3O4=";
```
https://github.com/ocaml/ocaml-lsp/tree/1.9.1

File: pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix (hunk 4 above)
Comment (symphorien) [reply pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:19]: 1.10.6 silently dropped support for ocaml 4.13 and only targets 4.14. It's not nice for a patch release.

File: pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix (hunk 6 above)
Comment (symphorien) [reply pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:23]: 1.9.1 targets ocaml 4.13 only.

**Review by superherointj — APPROVED** [review]

File: pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix (hunk 2 above)
Comment (symphorien) [reply pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix]: applied

File: pkgs/top-level/ocaml-packages.nix (hunk 1 above)
Comment (symphorien) [reply pkgs/top-level/ocaml-packages.nix]: I reordered the list alphabetically.

**Review by superherointj — APPROVED** [review]

**Comment by romildo** [comment]
Result of `nixpkgs-review pr 170142` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>stog</li>
  </ul>
</details>

**Comment by anmonteiro** [comment]
@symphorien thanks for upstreaming this! I think you just missed `dune-rpc-lwt`

**Comment by superherointj** [comment]
@anmonteiro As it is already merged. If you know the proper solution, feel free to open a new PR.

**Comment by symphorien** [comment]
What do you mean by missed? Honestly I won't have time to fix it soon, anyway.

**Pull request merged**